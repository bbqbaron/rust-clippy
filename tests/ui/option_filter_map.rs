fn odds_out(x: i32) -> Option<i32> {
    if x % 2 == 0 {
        Some(x)
    } else {
        None
    }
}

fn main() {
    let evens: Vec<i32> = vec![1, 2, 3]
        .into_iter()
        .map(odds_out)
        .filter(Option::is_some)
        .map(Option::unwrap)
        .collect();
}
