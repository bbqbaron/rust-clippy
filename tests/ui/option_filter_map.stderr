error: `filter` for `Some` followed by `unwrap`
  --> $DIR/option_filter_map.rs:10:27
   |
LL |       let evens: Vec<i32> = vec![1, 2, 3]
   |  ___________________________^
LL | |         .into_iter()
LL | |         .map(odds_out)
LL | |         .filter(Option::is_some)
LL | |         .map(Option::unwrap)
   | |____________________________^
   |
   = note: `-D clippy::option-filter-map` implied by `-D warnings`
   = help: consider using `flatten` instead

error: aborting due to previous error

